custom/icds_reports/static/js/directives/additional-filter/additional-filter.directive.js

```diff
@@ -58,15 +58,17 @@ function AdditionalFilterController($scope, $location, $uibModal, storageService
     vm.selectedGender = $location.search()['gender'] !== void(0) ? $location.search()['gender'] : '';
     vm.selectedAge = $location.search()['age'] !== void(0) ? $location.search()['age'] : '';
     var filtersObjects = [];
-    // eg:vm.filters = ['gender', 'age']
-    // this array has filters that are not to be shown. so if 'gender' is not in array, it can be shown.
-    if (vm.filters && vm.filters.indexOf('gender') === -1) {
-        vm.showGenderFilter = true;
-        filtersObjects.push({ label: 'Gender', value: vm.selectedGender });
-    }
-    if (vm.filters && vm.filters.indexOf('age') === -1) {
-        vm.showAgeFilter = true;
-        filtersObjects.push({ label: 'Age', value: vm.selectedAge });
+    vm.$onInit = function () {
```

Comment:
life cycle methods?

Reply:
yes. Accessing/changing the bindings of the component only  after they are initialized